syntax = "proto3";

package suggestions;

service SuggestionsService {
  rpc Suggestions (SuggestionRequest) returns (SuggestionResponse);
}

message VectorClock {
  map<string, int32> events = 6;
}

message Item {
  string name = 1;
  int32 quantity = 2;
}

message SuggestionRequest {
  string orderId = 1;
  repeated Item items = 3;
}

message Book {
  string bookId = 1;
  string title = 2;
  string author = 3;
}

message SuggestionResponse {
  string orderId = 1;
  VectorClock vectorClock = 2;
  repeated Book suggestedBooks = 3;
}

message ErrorResponse {
  string code = 1;
  string message = 2;
}

